src/systems/userinput/userinput-debug.js

```diff
@@ -3,11 +3,44 @@ AFRAME.registerSystem("userinput-debug", {
   active: true,
 
   tick() {
-    if (!this.active) {
-      return;
-    }
+    if (!this.active) return;
+
     const userinput = AFRAME.scenes[0].systems.userinput;
+
+    if (this.userinputFrameStatus) {
+      const replacer = (k, v) => {
+        if (typeof v === "number") {
+          return `${v >= 0 ? "+" : ""}${v.toFixed(3)}`;
+        }
+        return v;
+      };
+      const pathsOutput = [];
+      const { frame } = userinput;
+      for (const key in frame) {
+        if (!frame.hasOwnProperty(key)) continue;
+        if (!["/wmr-user", "/actions/"].some(x => key.startsWith(x))) continue;
+        let val = JSON.stringify(frame[key], replacer);
+        if (val) val = val.replace(/{"(\w{3,})":/g, '{\n  "$1":').replace(/,"(\w{3,})":/g, ',\n  "$1":');
+        pathsOutput.push(`${key} -> ${val}`);
+      }
+      this.userinputFrameStatus.textContent = pathsOutput.join("\n");
+    }
+
     if (userinput.get(paths.actions.logDebugFrame)) {
+      if (!this.userinputFrameStatus) {
```

Comment:
Nifty. I did a little experiment with rstats (the thing aframe uses for perf display) to draw graphs of all the values over time. Was pretty cool, but needed work to be useful. This is a good start. I really like how debuggable this system should be.